### DISCUSSION TITLE
Same Traceid for multiple requests

### DISCUSSION THREAD

**Discussion by shravani154** [open-telemetry/opentelemetry-java-instrumentation#5714] [Q&A]
Hello, 
I have an application which is continuously consuming messages from Azure event hub using spring cloud streams. I have integrated Opentelemetry using auto-instrumentation  and **I see all the requests having the same traceid**. Is there anything that I could add to get a different trace id for each consumer request?

**Comment by trask** [comment]
hi @shravani154! are you consuming messages from Azure event hub via kafka? has traceid been propagated from the producer side?

**Reply by shravani154** [reply]
Yes I am consuming messages from Azure event hub via kafka. The trace ids are not propagated from producer side, they are getting generated in the consumer application.

**Reply by trask** [reply]
I would be happy to look into this if you are able to create a repository that has a small repro that I can use to run and troubleshoot

**Reply by shravani154** [reply]
@Bean
    public Consumer<String> traces() {    return s -> log.info"(received message successfully",s);
   
    }
I am receiving the same trace id for every consumer record (until it finishes the consuming process) but I expect a different trace id for each consumer record.

**Reply by trask** [reply]
@shravani154 can you post a complete/runnable repro? it really helps to speed up the investigation process on our side, thx!

**Reply by shravani154** [reply]
Hello Trask... Unfortunately I am not able to expose the code I am working on... But here is the issue

Using the spring cloud streams, I have a consumer bean which listens to the eventhub...it seems that all the messages which are consumed are running on the same thread and hence we are getting same trace id for all the consumed messages. Ideally I need to have a different trace id for each message getting produced... Here I am using java agent for generating the traces... So How can I achieve generating different traceid?

**Reply by trask** [reply]
> Unfortunately I am not able to expose the code I am working on

can you create a minimal repro of the issue? it's very hard to investigate/troubleshoot this kind of issue without something that we can run and debug ourselves

**Reply by shravani154** [reply]
Can we set a custom trace id for the current span generated by java agent(auto instrumentation)?

**Reply by trask** [reply]
hi @shravani154! I think I realized what's probably going on here.

I sent PR #5826 to try to address.
